cellforest/structures/Counts.py

```diff
@@ -78,29 +86,182 @@ def hstack(self, others: Union["Counts", Iterable["Counts"]]):
+        axis: int = 0,
+        labels: Optional[Union[pd.Series, list]] = None,
+        ax: Axes = None,
+        **kwargs,
+    ) -> Axes:
         """
-        Plots histogram along specified axis, optionally, stratified by label.
+        Plots histogram along specified axis, optionally, stratified by label
         Args:
-            agg: aggregation function for opposite axis (e.g. sum, min, mean, var, etc.)
+            agg: name of aggregation function for opposite axis (e.g., "std"); all options: `self._SUPPORTED_AGG_FUNCS`
             axis: axis along which to create histogram, with `agg` applied to other axis
-            labels: cell or gene category labels by which to stratify plot
+            labels: list or pd.Series of cell or gene category labels by which to stratify plot
+            ax: matplotlib pyplot or axes object which defines the plot
+            kwargs: keyword arguments for plt.hist()
 
         Returns:
-            hist: histogram
+            ax: histogram
+
+        Examples:
+            >>> rna = Counts.from_cellranger("../tests/data/v3_gz/sample_1")  # load Counts matrix
+            >>> half_of_cells = rna._matrix.shape[0] // 2
+            >>> labels = ["sample_1"] * half_of_cells + ["sample_2"] * half_of_cells  # mock cell labels for 2 samples
+            >>> fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 6))  # figure with 1x2 axes
+            >>> rna.hist("sum", axis=0, ax=ax1, labels=labels, bins=30, alpha=0.5, histtype='step')  # plot on 1st axes object
+            >>> rna.hist("std", axis=0, ax=ax2)  # plot on 2nd axes object
+            >>> fig.show()  # display figure state
+
+            >>> rna = Counts.from_cellranger("../tests/data/v3_gz/sample_1")  # load Counts matrix
+            >>> rna.hist("sum", axis="genes", color="#7eaa53", bins=30)  # plot on currect (or newly created) axes object
+            >>> plt.show()  # display current figure with axes
         """
-        # TODO: QUEUE
-        raise NotImplementedError()
+
+        axis = self._get_numeric_axis(axis)
+        labels = self._get_agg_labels(labels, axis)
+
+        cells_axis = axis == 0  # bool for aggregated axis' name
+        cs_matrix = (
+            self._matrix.tocsr() if cells_axis else self._matrix.tocsc()
+        )  # convert to compressed sparse column/row for fast arithmetics
+
+        ax = ax or plt.gca()  # use defined or get current axes
+        for label in set(labels):
+            where_label = np.where(np.array(labels) == label)[0]
+            matrix_slice = cs_matrix[where_label, :] if cells_axis else cs_matrix[:, where_label]
+            rna_agg = self._agg_apply(matrix_slice, agg=agg, axis=axis)
+            ax.hist(rna_agg, label=label, **kwargs)
+
+        x_label = "transcript count" if cells_axis else "cell count"
+        title = x_label + " " + ("per cell" if cells_axis else "per gene")
+        ax.set_title("{} of ".format(agg) + title)
+        ax.set_ylabel("quantity")
```

Comment:
Would be good to calculate a more clear label here as you did for the x axis

Reply:
Do you mean a customized `ylabel`?